### PR TITLE
[Minor Opt] Make miners cache mined item

### PR DESCRIPTION
**Pull request by Phlosioneer**
Saves two chunk+tile lookups per update, and one chunk+tile lookup per draw.

### CODE REVIEW

**Comment by tobspr** [comment]
This will break existing savegames, it would be required to write a migration then (Check out how 1003 -> 1004 works for example. simply search for `1004` in the code). Looks good otherwise!

**Comment by tobspr** [comment]
I would recommend not to serialize/save this - Instead make the miner system compute it whenever an entity gets added (root.signals.entityAdded)

**Comment by Phlosioneer** [comment]
Yeah, I learned I could avoid serialization on the later PR's. It makes more sense here.

File: src/js/game/systems/miner.js [hunk 1]
```diff
@@ -114,18 +109,17 @@ export class MinerSystem extends GameSystemWithFilter {
...
+                        continue;
+                    }
 
-                    const lowerLayerItem = this.root.map.getLowerLayerContentXY(
-                        staticComp.origin.x,
-                        staticComp.origin.y
-                    );
-
-                    if (lowerLayerItem) {
+                    if (minerComp.cachedMinedItem) {
```
Comment (tobspr) [inline src/js/game/systems/miner.js:120]: This if is unnecessary, since you continue above. I can remove it myself tho

**Comment by tobspr** [comment]
Thank you! Will be in 1.1.16

**Pull request merged**